src/platform/w3dengine/client/w3dbridgebuffer.cpp

```diff
@@ -0,0 +1,788 @@
+
+            if (next == nullptr || !next->Get_Flag(FLAG_BRIDGE_POINT2)) {
+                captainslog_debug("Missing second bridge point.  Ignoring first.");
+            }
+
+            if (next == nullptr) {
+                break;
+            }
+
+            if (next->Get_Flag(FLAG_BRIDGE_POINT2)) {
+                Vector3 point1;
+                Vector3 point2;
+                point1.Set(obj->Get_Location()->x, obj->Get_Location()->y, 0.0f);
+                point1.Z = g_theTerrainRenderObject->Get_Height_Map_Height(point1.X, point1.Y, nullptr) + 0.25f;
+                point2.Set(next->Get_Location()->x, next->Get_Location()->y, 0.0f);
+                point2.Z = g_theTerrainRenderObject->Get_Height_Map_Height(point2.X, point2.Y, nullptr) + 0.25f;
+
+                Add_Bridge(point1, point2, obj->Get_Name(), terrain_logic, obj->Get_Properties());
+            }
+        }
+    }
+
+    if (terrain_logic != nullptr) {
+        terrain_logic->Update_Bridge_Damage_States();
+    }
+}
+
+void W3DBridgeBuffer::Add_Bridge(
+    Vector3 from_loc, Vector3 to_loc, Utf8String name, W3DTerrainLogic *terrain_logic, Dict *props)
+{
+    if (m_numBridges < MAX_BRIDGES && m_initialized) {
+        m_bridges[m_numBridges].Init(from_loc, to_loc, name);
+
+        if (m_bridges[m_numBridges].Load(BODY_PRISTINE)) {
+            if (terrain_logic != nullptr) {
+                BridgeInfo info;
+                m_bridges[m_numBridges].Get_Bridge_Info(&info);
+                info.bridge_index = m_numBridges;
+                terrain_logic->Add_Bridge_To_Logic(&info, props, name);
+            }
+
+            m_numBridges++;
+        }
+    }
+}
+
+void W3DBridgeBuffer::Update_Center(CameraClass *camera, RefMultiListIterator<RenderObjClass> *lights)
+{
+    Cull(camera);
+
+    if (m_anythingChanged || !m_curNumBridgeIndices) {
+        Load_Bridges_In_Vertex_And_Index_Buffers(lights);
+    }
+
+    m_updateVis = false;
+}
+
+void W3DBridgeBuffer::Draw_Bridges(CameraClass *camera, bool wireframe, TextureClass *cloud_texture)
+{
+    if (g_theTerrainLogic != nullptr) {
+        for (int i = 0; i < m_numBridges; i++) {
+            m_bridges[i].Set_Enabled(false);
+        }
+
+        bool draw = false;
+
+        for (Bridge *bridge = g_theTerrainLogic->Get_First_Bridge(); bridge; bridge = bridge->Get_Next()) {
+            BridgeInfo info;
+            bridge->Get_Bridge_Info(&info);
+
+            if (info.bridge_index >= 0 && info.bridge_index < m_numBridges) {
+                m_bridges[info.bridge_index].Set_Enabled(true);
+
+                if (m_bridges[info.bridge_index].Get_Damage_State() != info.cur_damage_state) {
+                    BodyDamageType cur_damage_state = m_bridges[info.bridge_index].Get_Damage_State();
+                    m_bridges[info.bridge_index].Set_Damage_State(info.cur_damage_state);
+
+                    if (!m_bridges[info.bridge_index].Load(info.cur_damage_state)) {
+                        draw = true;

```

Comment:
was above the if, if this is a bugfix comment as such

Reply:
fixed.